### PR TITLE
`pythagorean-triplet` -- Fix test and propose new example solution

### PR DESCRIPTION
**Pull request by jiegillet**
I found a typo in the test, that's the main thing.

I'm also sharing my solution that passes the tests in 0.06s versus 10s for the existing one, but I know that the purpose of an example solution is to prove it can be done and not necessarily be efficient, so I don't feel too strongly about making that change.

### CODE REVIEW

**Comment by angelikatyborska** [comment]
> I'm also sharing my solution that passes the tests in 0.06s versus 10s for the existing one

WHAT 😮 what are you, a computer witch/wizard? This is amazing!

**Comment by angelikatyborska** [comment]
Your solution is so superior to the old one that it doesn't even need processes to parallelize calculations :D

Now I need to update "practices" because they recommend "tasks" and "list-comprehensions", but that is obviously leading students in the wrong direction.

I think we will need to write mentoring notes for this exercise to explain https://en.wikipedia.org/wiki/Tree_of_primitive_Pythagorean_triples and how that's necessary for a fast solution

**Comment by jiegillet** [comment]
Hahaha, sorry for the extra work :)

How do mentoring notes work? Do you mean hints?

**Comment by angelikatyborska** [comment]
> Do you mean hints?

No, I do not mean hints, but now that you mention it, that's an even better idea!

Hints would be a document in the exercise's directory (documentation: https://github.com/exercism/docs/blob/main/building/tracks/practice-exercises.md#file-docshintsmd) and each student would see it while solving the exercise. Let's add a hint for this 🙂 

But to also answer the question, mentoring notes are stored in https://github.com/exercism/website-copy/tree/main/tracks/elixir/exercises and they are documents aimed at mentors. They usually list a good solution to an exercise (as opposed to the example solutions in this repository, which are just supposed to work, they don't need to be good). They also list some hints about what to pay attention to when mentoring a student in this exercise, popular mistakes that people make etc.

For mentoring notes to be useful in this situation, the student would need to request mentoring, and then the mentor would need to realize that the solution is too slow and look up the mentoring notes. That's a lot of extra steps... hints sound like a better idea.

**Comment by jiegillet** [comment]
I added a hint, I hope it's not too humorous.
I'll let you add the mentoring notes if that's OK, because I'm not a mentor (although I'm considering it).

File: exercises/practice/pythagorean-triplet/.docs/hints.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+## General
+
+- If checking through all sorts of triangles is too slow, consider that [right triangles grow on trees](triples).
```
Comment (angelikatyborska) [inline exercises/practice/pythagorean-triplet/.docs/hints.md:3]: I love it 😂

Comment (jiegillet) [inline exercises/practice/pythagorean-triplet/.docs/hints.md:3]:
```suggestion
- If checking through all sorts of triangles is too slow, consider that [right triangles grow on trees](triples).
```

**Comment by angelikatyborska** [comment]
I pushed a small change to prerequisites and practices so that they make sense with the clearly superior new example.

I think you would be a great mentor! You should try it sometime. Trying to figure out the correct advice to give to students is quite similar to doing code reviews.

**Comment by angelikatyborska** [comment]
Oh, and I think the hint is enough. If we have the hint, hopefully we don't need mentor notes.

**Review by angelikatyborska — APPROVED** [review]

**Pull request merged**